rfc-138-enable-brotli-compression.md

```diff
@@ -0,0 +1,88 @@
+
+![Visual progress of the homepage using both compression methods.](rfc-138/visual-progress-homepage.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-homepage.png)
+
+### Coronavirus page
+
+The visual progress graph for this page starts off worse by 300ms, but quickly catches up and the viewport for Brotli completes rendering 200ms before gzip:
+
+![visual progress graph for the covid page.](rfc-138/visual-progress-covid.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-coronavirus.png)
+
+Showing a 5.53% reduction.
+
+### Bank holidays
+
+The visual progress of the bank holidays has improved by approximately 200ms as can be seen in the visual progress graph:
+
+![Visual progress of the bank holidays using both compression methods.](rfc-138/visual-progress-bank-holidays.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-bank-holidays.png)
+
+Showing a 4.60% reduction.
+
+### Search page
+
+The visual progress of the search page has improved by approximately 300ms initially, 200ms at the end. As can be seen in the visual progress graph:
+
+![Visual progress of the bank holidays using both compression methods.](rfc-138/visual-progress-search.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-search.png)
+
+Showing a 5.56% reduction.
+
+### Summary 
+
+In all the pages tested across the frontend apps (17 in total) we see between 3-6% reduction in the number of bytes a browser is having to download over the network. This saving is being converted to improvements in the visual metrics a users is seeing in the browser.
+
+## Proposal
+
+The proposal is to enable Brotli compression at the edge using the [Fastly Brotli Compression LA](https://www.fastly.com/release-notes/q3-2020#brotli). Browsers that support Brotli will receive these versions of the files. Any browsers that don't will receive the gzip version of the file. This can therefore be applied using the progressive enhancement methodology. In doing so we MUST update the configuration VCL so as to automate this process. During testing it was simply enabled via the Fastly user interface.
+
+We SHOULD also purge the static cache once completed is it was found during testing that certain Fastly PoP's continued to serve the uncompressed version of the file since there was a mismatch between `Accept-Encoding` header from the browser and `Content-Encoding` version available on the CDN. Once the cache was purged this issue was rectified. This issue will rectify itself over time anyway even without the manual purging since Rails inserts an MD5 fingerprint into the filename of each file. Once a file is changed it will have a new URL, thus invalidating the old cache.
```

Comment:
```suggestion
We SHOULD also purge the static cache once completed as it was found during testing that certain Fastly PoP's continued to serve the uncompressed version of the file since there was a mismatch between `Accept-Encoding` header from the browser and `Content-Encoding` version available on the CDN. Once the cache was purged this issue was rectified. However, this issue would rectify itself over time even without the manual purging, since Rails inserts an MD5 fingerprint into the filename of each file. Once a file is changed it will have a new URL, thus invalidating the old cache.
```